conda/connection.py

```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
             return None
 
     @staticmethod
-    def is_binstar_url(url):
+    def is_binstar_url_needing_token(url):
         urlparts = urlparse(url)
+        token, path = split_token(urlparts.path)
+        if token:  # url already has token
+            return False
         return (urlparts.scheme in ('http', 'https') and
                 any(urlparts.hostname.endswith(bh) for bh in context.binstar_hosts))
```

Comment:
> should we explicitly document that

Yes, in general I could have used more documentation/specification about intended behavior.  For example, just learned that if we use bare IP addresses for hostname, we use `/api` and `/conda` as part of the path.